src/utils/mod.rs

```diff
@@ -107,9 +107,27 @@ where
     F: FnOnce() -> Result<R> + Send + 'static,
     R: Send + 'static,
 {
-    tokio::task::spawn_blocking(f)
+    let result = tokio::task::spawn_blocking(f)
         .await
-        .context("failed to join thread")?
+        .context("failed to join thread")?;
+
+    if cfg!(debug_assertions) {
+        // Since `AxumNope` can also be silently converted into
+        // `anyhow::Error`, someone accidentally using this method
+        // instead of `web::error::spawn_blocking_web` can lead to
+        // a more specific AxumNope error with a specific status code
+        // being silently converted into an anyhow::Error, so a server
+        // error with status 500.
```

Comment:
We could have a custom `impl From<anyhow::Error> for AxumNope` that attempts to downcast and just return a wrapped `AxumNope`.

Reply:
good idea! 

I updated the code, similarly checking for an old iron-`Nope` for the migration time.